### ISSUE TITLE
"lines-around-comment" and "padded-blocks" will be conflict 

### ISSUE THREAD

**Issue by dawnwish** [eslint/eslint#6526]
**What version of ESLint are you using?**
2.13.0

**What parser (default, Babel-ESLint, etc.) are you using?**
Espree

**Please show your full configuration:**

``` js
{
    "root": true,
    "extends": "airbnb",
    "env": {
        "browser": true,
        "node": true
    },
    "rules": {
        "lines-around-comment": [2, { "beforeBlockComment": true, "beforeLineComment": true }],
        "padded-blocks": [2, "never"]
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

``` js
if (!accountInfo) {

    // xxxxxxx
    console.log('xxx');
}
```

**What actually happened? Please include the actual, raw output from ESLint.**
if i add the blank line before the comment, eslint show 
`padded-blocks Block must not be padded by blank lines.`
but if i remove it, eslint show
`lines-around-comment Expected line before comment`

**Comment by vitorbal** [comment]
Thanks, @dawnwish! Just verified that this happens. I think it should be considered a bug that one rule “negates” the other like this, but I would like to have some more input from other members of the team before marking it as "accepted".

**Comment by ilyavolodin** [comment]
Hmm.. this is a problem. Rules shouldn't conflict like that with each other. However, I'm not sure how we can fix this. Rules also can't know about each other. Each rule independently acts correctly in this case, but when you put them together you get into unfixable situation. The only thing that comes to my mind is to change `padded-block` to treat comments as empty lines. But that would still not fix all possible conflicts anyways. The following configuration:
```js
    "rules": {
        "lines-around-comment": [2, { "afterBlockComment": true, "afterLineComment": true }],
        "padded-blocks": [2, "never"]
    }
```
And this code:
```js
function foo() {
  console.log('bar');
  // some comment

}
```

is still going to create conflicts.
I'm not 100% sure we can do anything about this case.

**Comment by platinumazure** [comment]
Doesn't `lines-around-comment` have options which allow/forbid padding at the start or end of a block (overriding the general rule)?

It's unfortunate that there exist configurations which can cause conflict, but there are also configurations which do not cause conflict and do what the user wants. We just need to encourage the user to use those configurations.

**Comment by mysticatea** [comment]
Personally, I'd like `lines-around-comment` to ignore comments at the head of a block and the last of a block.
The behavior is same as `newline-after-var` and `newline-before-return`.

**Comment by kaicataldo** [comment]
Don't `lines-around-comment`'s `allowBlockStart` and `allowBlockEnd` options cover this incompatibility?

**Comment by kaicataldo** [comment]
I'm going to close this as I believe this is covered by `lines-around-comment`'s `allowBlockStart` and `allowBlockEnd` options and there hasn't been any activity in quite a long time.

**Comment by dawnwish** [comment]
@kaicataldo  yes, `allowBlockStart ` and `allowBlockEnd ` can resolve this problem,  I didn't noticed these options, thank you!

**Comment by jrencz** [comment]
Just for the record: I opened https://github.com/eslint/eslint/issues/9653 which describes a bug in "lines-around-comment".

"lines-around-comment" and "padded-blocks" treat switch block different. With the following configuration:

```json
{
  "rules": {
    "lines-around-comment": ["error", {
      "beforeBlockComment": true,
      "beforeLineComment": true,
      "allowBlockStart": true,
      "allowBlockEnd": false,
      "allowObjectStart": true,
      "allowObjectEnd": false,
      "allowArrayStart": true,
      "allowArrayEnd": false,
    }],
   "padded-blocks": ["error", "never"]
  }
}
```

Here's a demo showing where's the problem using the exact configuration as above.
<a href="https://eslint.org/demo/#eyJ0ZXh0IjoiLyplc2xpbnQgbGluZXMtYXJvdW5kLWNvbW1lbnQ6IFtcImVycm9yXCIsIHsgXCJiZWZvcmVCbG9ja0NvbW1lbnRcIjogdHJ1ZSwgXCJiZWZvcmVMaW5lQ29tbWVudFwiOiB0cnVlLCBcImFsbG93QmxvY2tTdGFydFwiOiB0cnVlLCBcImFsbG93QmxvY2tFbmRcIjogZmFsc2UsIFwiYWxsb3dPYmplY3RTdGFydFwiOiB0cnVlLCBcImFsbG93T2JqZWN0RW5kXCI6IGZhbHNlLCBcImFsbG93QXJyYXlTdGFydFwiOiB0cnVlLCBcImFsbG93QXJyYXlFbmRcIjogZmFsc2UgfV0gcGFkZGVkLWJsb2NrczogW1wiZXJyb3JcIiwgXCJuZXZlclwiXSovXG5cbnZhciBhID0gZnVuY3Rpb24gKCkge1xuICAvLyBGb29cbiAgLy8gQmFyXG4gIFxuICB2YXIgYiA9IDE7XG59XG5cbnN3aXRjaCAoZm9vKSB7XG4gIC8vIEZvb1xuICAvLyBCYXJcbiAgXG4gIGNhc2UgMToge1xuICAgIC8vIEZvb1xuICB9XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo1LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19" target="_blank"><img src="https://www.google.com/webpagethumbnail?c=41&r=4&s=300&d=https://eslint.org/demo/%23eyJ0ZXh0IjoiLyplc2xpbnQgbGluZXMtYXJvdW5kLWNvbW1lbnQ6IFtcImVycm9yXCIsIHsgXCJiZWZvcmVCbG9ja0NvbW1lbnRcIjogdHJ1ZSwgXCJiZWZvcmVMaW5lQ29tbWVudFwiOiB0cnVlLCBcImFsbG93QmxvY2tTdGFydFwiOiB0cnVlLCBcImFsbG93QmxvY2tFbmRcIjogZmFsc2UsIFwiYWxsb3dPYmplY3RTdGFydFwiOiB0cnVlLCBcImFsbG93T2JqZWN0RW5kXCI6IGZhbHNlLCBcImFsbG93QXJyYXlTdGFydFwiOiB0cnVlLCBcImFsbG93QXJyYXlFbmRcIjogZmFsc2UgfV0gcGFkZGVkLWJsb2NrczogW1wiZXJyb3JcIiwgXCJuZXZlclwiXSovXG5cbnZhciBhID0gZnVuY3Rpb24gKCkge1xuICAvLyBGb29cbiAgLy8gQmFyXG4gIFxuICB2YXIgYiA9IDE7XG59XG5cbnN3aXRjaCAoZm9vKSB7XG4gIC8vIEZvb1xuICAvLyBCYXJcbiAgXG4gIGNhc2UgMToge1xuICAgIC8vIEZvb1xuICB9XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo1LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19&a=AIYkKU8Fz1HjUawYyNWx8hgeLLkrG2RoFw"></a>
Neither adding nor removing the line above comment inside switch helps. Either rule is unhappy in that case
